controllers/vaultdynamicsecret_controller.go

```diff
@@ -591,6 +682,57 @@ func getRotationDuration(o *secretsv1beta1.VaultDynamicSecret) time.Duration {
 	return d
 }
 
+// vaultClientCallback requests reconciliation of all VaultDynamicSecret instances that were synced with Client
+func (r *VaultDynamicSecretReconciler) vaultClientCallback(ctx context.Context, c vault.Client, _ error) {
+	logger := log.FromContext(ctx).WithName("vaultClientCallback")
+
+	cacheKey, err := c.GetCacheKey()
+	if err != nil {
+		// should never get here
+		logger.Error(err, "Invalid cache key, skipping syncing of VaultDynamicSecret instances")
+		return
+	}
+
+	logger = logger.WithValues("cacheKey", cacheKey, "controller", "vds")
+	var l secretsv1beta1.VaultDynamicSecretList
+	if err := r.Client.List(ctx, &l, client.InNamespace(
+		c.GetCredentialProvider().GetNamespace()),
+	); err != nil {
+		logger.Error(err, "Failed to list VaultDynamicSecret instances")
+		return
+	}
+
+	if len(l.Items) == 0 {
+		return
+	}
+
+	reqs := map[SyncRequest]empty{}
+	for _, o := range l.Items {
+		if o.Status.VaultClientMeta.CacheKey == cacheKey.String() {
+			_, delay := computeMaxJitterDuration(vdsMaxClientCallbackDelayForJitter)
+			req := SyncRequest{
+				Delay:        delay,
+				RequeueOnErr: true,
+				Request: reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: o.GetNamespace(),
+						Name:      o.GetName(),
+					},
+				},
+			}
+			if _, ok := reqs[req]; !ok {
+				reqs[req] = empty{}
+				logger.V(consts.LogLevelDebug).Info("Calling Sync()",
+					"request", req, "delay", delay)
+				r.SyncRegistry.Add(req.NamespacedName)
+				if _, err := r.SyncController.Sync(ctx, req); err != nil {
+					logger.Error(err, "Sync failed", "request", req)
+				}
```

Comment:
Remind me the purpose of the SyncRegistry here? Is it just for setting the syncReason in `r.SyncController.Sync()`?

Reply:
It is used to force a sync of a specific resource. It is pretty well documented here: https://github.com/hashicorp/vault-secrets-operator/blob/e7a26872a4b254fe9a1b5827b4d01c2f50f27454/controllers/registry.go#L181 - let me know if you think we need to enhance the docs.